python/triton/runtime/autotuner.py

```diff
@@ -184,12 +207,34 @@ def run(self, *args, **kwargs):
             if key not in self.cache:
                 # prune configs
                 used_cached_result = False
+                # Step 1: Parallel compilation
                 pruned_configs = self.prune_configs(kwargs)
+                compile_start = time.time()
+                compiled_configs = []
+                with ThreadPoolExecutor() as executor:
+                    futures = {
+                        executor.submit(self._compile, *args, config=config, **kwargs): config
+                        for config in pruned_configs
+                    }
+                    for future in as_completed(futures):
+                        config, success = future.result()
+                        if success:
+                            compiled_configs.append(config)
+                compile_end = time.time()
+
+                # Step 2: Linear benchmarking
                 bench_start = time.time()
-                timings = {config: self._bench(*args, config=config, **kwargs) for config in pruned_configs}
+                timings = {}
+                for config in compiled_configs:
+                    timings[config] = self._bench(*args, config=config, **kwargs)
```

Comment:
I think this part could be left as-is:
```suggestion
                timings = {config: self._bench(*args, config=config, **kwargs) for config in pruned_configs}
```